src/message_dao_test.py

```diff
@@ -0,0 +1,115 @@
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(message_dao, 'parse_message_from_str')
+    def test_get(self, mock_parse_msg, mock_check_output):
+        # Testing searching for a message id that's not in DAO
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get")
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        mock_parse_msg.return_value = email
+        self.mock_cursor.fetchone.return_value = (email.archive_hash, email.change_id)
+        self.assertEqual(email, self.dao.get('fake_message_id'))
+        self.assertEqual(2, self.mock_execute.call_count) 
+        self.reset_mocks()
+
+        # Testing searching for a message id that's not in DAO
+        self.mock_cursor.fetchone.return_value = None
+        self.assertIsNone(self.dao.get('not_in_dao'))
+        self.mock_execute.assert_called_once()
+
+    def test_size(self):
+        # Testing size of empty dao
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get")
+        self.mock_cursor.fetchone.return_value = (0,)
+        self.assertEqual(0, self.dao.size())
+        self.mock_execute.assert_called_once()
+        self.reset_mocks()
+
+        # Testing size of one dao
+        self.mock_cursor.fetchone.return_value = (1,)
+        self.assertEqual(1, self.dao.size())
+        self.mock_execute.assert_called_once()
+        self.reset_mocks()
+
+        # Testing size of many dao
+        self.mock_cursor.fetchone.return_value = (10,)
+        self.assertEqual(10, self.dao.size())
+        self.mock_execute.assert_called_once()
+
+    def test_store_hash(self):
+        # Testing storing hash when it's not already in State table
+        self.dao.store_last_hash("some_hash")
+        sql_text = "REPLACE INTO States VALUES (%s, %s)"
+        self.mock_execute.assert_called_once_with(sql_text, mock.ANY)
+        self.mock_commit.assert_called_once()
+        self.reset_mocks()
+
+        # Testing storing hash when it's already in State table
+        self.dao.store_last_hash("some_hash")
+        self.mock_execute.assert_called_once()
+        self.mock_commit.assert_called_once()
+
+    def test_get_hash(self):
+        # Testing searching for hash when it's not already in State table
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get_hash")
+        self.mock_cursor.fetchone.return_value = None
+        self.assertEqual(message_dao.EPOCH_HASH, self.dao.get_last_hash())
+        self.mock_execute.assert_called_once()
+        self.reset_mocks()
+
+        # Testing searching for hash when it's already in State table
+        self.mock_cursor.fetchone.return_value = ("fake_hash",)
+        self.assertEqual("fake_hash", self.dao.get_last_hash())
+        self.mock_execute.assert_called_once()
+
+if __name__ == '__main__':
+    # TODO(@lenhard): Issue with Google's Connector that causes segmentation fault
```

Comment:
nit: we usually would write TODO(email), so TODO(lenhard@google.com)

Reply:
Got it; I was just going off of the TODO in main.py.